ui/app/pages/swaps/view-quote/view-quote-price-difference.js

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from 'react'
+
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import BigNumber from 'bignumber.js'
+import { useEthFiatAmount } from '../../../hooks/useEthFiatAmount'
+import { I18nContext } from '../../../contexts/i18n'
+
+import ActionableMessage from '../actionable-message'
+import Tooltip from '../../../components/ui/tooltip'
+
+export default function ViewQuotePriceDifference(props) {
+  const t = useContext(I18nContext)
+
+  const { usedQuote, sourceTokenValue, destinationTokenValue } = props
+
+  const priceSlippageFromSource = useEthFiatAmount(
+    usedQuote?.priceSlippage?.sourceAmountInETH || 0,
+  )
+  const priceSlippageFromDestination = useEthFiatAmount(
+    usedQuote?.priceSlippage?.destinationAmountInEth || 0,
+  )
+  const priceSlippageUnknownFiatValue =
+    !priceSlippageFromSource || !priceSlippageFromDestination
+
+  if (!usedQuote || !usedQuote.priceSlippage) {
+    return null
+  }
+
+  const { priceSlippage } = usedQuote
+
+  let priceDifferencePercentage = 0
+  if (priceSlippage.ratio) {
+    priceDifferencePercentage = parseFloat(
+      new BigNumber(priceSlippage.ratio, 10)
+        .minus(1, 10)
+        .times(100, 10)
+        .toFixed(2),
+      10,
+    )
+  }
+
+  let priceDifferenceTitle = ''
+  let priceDifferenceMessage = ''
+  if (priceSlippage.calculationError) {
+    // An error was returned by the API; show it to the user
+    priceDifferenceMessage = priceSlippage.calculationError
```

Comment:
This could be a bad assumption, but if there is a calculation error, we show it to the user directly.  Thoughts?

Reply:
I think in this case we should just show the "unknown impact" message.